src/noggit/texture_set.cpp

```diff
@@ -425,11 +426,40 @@ bool TextureSet::paintTexture(float xbase, float zbase, float x, float z, Brush*
             else
             {
               amaps[layer][offset] -= alpha_change * (amaps[layer][offset] / sum_other_alphas);
+
+            }
+
+            if (amaps[layer][offset] > 1.0f)
+                amaps[layer][offset] = 1.0f;
+            else if (amaps[layer][offset] < 0.0f || isnan(amaps[layer][offset]))
```

Comment:
What is the operation making it nan, ` / sum_other_alphas`? If that's it, I'd suggest changing `sum_other_alphas = 1.f - current_alpha` above to `sum_other_alphas = std::max (1.f - current_alpha, std::numeric_limits<float>::epsilon())` (include `<limits>` if that's not yet the case), which would make the division always sane and never produce a NaN.

Then, here we can just `amaps[layer][offset] = std::min (1.f, std::max (0.f, amaps[layer][offset]));` to express what the condition is.